### DISCUSSION TITLE
How to build docs in GitHub without GitHub Actions or other CI/CD tools

### DISCUSSION THREAD

**Discussion by syeda-git** [mkdocs/mkdocs#2599] [Q&A]
Hello,

I am evaluating tools for my company. MkDocs has many advantages. One of the use cases is to edit the source files in GitHub itself and when you commit the changes, the GitHub Pages site is refreshed. So, GitHub automatically builds the changes and deploys it to GitHub pages in the background. My company's GitHub does not yet support GitHub Actions and we don't want invest time and effort now at setting up a CI/CD tool.

I heard that with Jekyll the above use case is possible. Maybe because GitHub Pages is also based on Jekyll. Is there a way to achieve this capability with MkDocs or MkDocs with Material?

I read these docs:
https://www.mkdocs.org/user-guide/deploying-your-docs/
https://squidfunk.github.io/mkdocs-material/publishing-your-site/

Let me know if you have questions.

Thanks...
Syed

**Comment by Andre601** [comment]
To be honest, I would just go and setup a GitHub Action. It doesn't take up much time and it's fairly easy to achieve.

Material for MkDocs even does have an explanation on how you do it: https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-github-actions

The most important part is just, that you need to trigger MkDocs' `gh-deploy` command to both build and push the site to the repository's `gh-pages` branch.

**Comment by oprypin** [comment]
GitHub just "bundles" Jekyll support, so it has an "unfair" advantage in that. If you have no way to run a site generator, then you're literally limited to whatever GitHub provides, that being either straight up the file browsing UI, or Jekyll. The only other thing I could think of is a "dual" approach where for preview you'd be relying on GitHub but what's deployed in the end is MkDocs. But the Markdown flavors have some large differences, so that'd just be a confusing mess.

**Comment by unites** [comment]
Sorry to resurrect the dead but I am in the same boat.  I don't have access to our companies Github instance to just "add" Actions, nor would the internal infra team want to incur a production change for my workflow alone. 

I am currently going to just try and write a cron job to run on a server that will do this but thought I would ask to see if you resolved this?

**Reply by sban2009** [reply]
same for me. our version is old and does not support self-hosted actions

**Reply by oprypin** [reply]
I really don't understand the premise of the question. The way I understand it, it's basically:
"How can I run some code without being able to run code?"

**Reply by sban2009** [reply]
@oprypin not sure about the original question but my org runs v3.1 that doesn't support github actions so we are manually pushing the built site to gh-pages without any automation. in the docs, there is no other method mentioned other than github actions. it would be helpful if other ci tools' methods were also there... [my discussion](https://github.com/mkdocs/mkdocs/discussions/2910)

**Reply by oprypin** [reply]
mkdocs' documentation actually doesn't mention GitHub Actions, instead it mentions `gh-deploy`, which just requires you to be able to run Python in any way.
https://www.mkdocs.org/user-guide/deploying-your-docs/

**Comment by inspire99** [comment]
@syeda-git / @oprypin  / @Andre601  / @unites / @sban2009 , I have added below which worked for me:

If GitHub Actions cannot be utilized, you can manually deploy your MkDocs project to GitHub Pages by following these steps:

1. **Local Environment Setup**:
   - Ensure that you have Python and MkDocs installed on your local machine.
   - Clone your GitHub repository to your local machine if it’s not already.

2. **Build the Project**:
   - Open a terminal and navigate to your MkDocs project directory.
   - Run the command `mkdocs build`. This will generate a `site` directory containing the built website.

3. **Prepare for Deployment**:
   - Create a new branch in your repository named `gh-pages` if it doesn’t already exist.
   - Delete everything in the `gh-pages` branch, as the new build will replace it.

```bash
git checkout --orphan gh-pages
git rm -rf .
```

4. **Move Built Files**:
   - Move the contents of the `site` directory to the root of your repository.

```bash
mv site/* .
rmdir site
```

5. **Commit and Push**:
   - Commit the changes and push the `gh-pages` branch to GitHub.

```bash
git add .
git commit -m "Deploy MkDocs to GitHub Pages"
git push origin gh-pages
```

6. **Configure GitHub Pages**:
   - Go to the settings of your GitHub repository.
   - Navigate to the Pages section and select the `gh-pages` branch as the source.

Your MkDocs documentation should now be accessible at `https://<username>.github.io/<repository>/`. Remember to repeat steps 2-5 every time you update your documentation.

**Reply by Andre601** [reply]
Or you just use the `gh-deploy` subcommand of mkdocs which does all these steps for you.

Also, don't tag me next time for things like this. Thanks.
